### PR TITLE
boards: efr32bg22_thunderboard: add PWM support

### PR DESCRIPTION
**Pull request by wkhadgar**
Add PWM support to the EFR32BG22 SoC. Also include the boards on the PWM tests, and update the documentation on the BG22 boards family.

### CODE REVIEW

File: tests/drivers/pwm/pwm_loopback/boards/efr32bg22_brd4184a.overlay [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+		       <&pwm1 0 0 PWM_POLARITY_NORMAL>;
+	};
+};
+
+&timer1 {
+	status = "okay";
+
+	pwm1: pwm {
+		status = "okay";
+		pin-location = <GECKO_LOCATION(0) GECKO_PORT_B GECKO_PIN(3)>;
```
Comment (fkokosinski) [inline tests/drivers/pwm/pwm_loopback/boards/efr32bg22_brd4184a.overlay:20]: It looks like you're using the legacy way of pinmuxing here. This is unfortunately no longer allow in Zephyr. You should switch to the Pin Control API.

File: tests/drivers/pwm/pwm_loopback/boards/efr32bg22_brd4184b.overlay [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+		       <&pwm1 0 0 PWM_POLARITY_NORMAL>;
+	};
+};
+
+&timer1 {
+	status = "okay";
+
+	pwm1: pwm {
+		status = "okay";
+		pin-location = <GECKO_LOCATION(0) GECKO_PORT_B GECKO_PIN(3)>;
```
Comment (fkokosinski) [inline tests/drivers/pwm/pwm_loopback/boards/efr32bg22_brd4184b.overlay:20]: Ditto

File: boards/arm/efr32_thunderboard/Kconfig.defconfig [hunk 3]
```diff
@@ -57,3 +57,7 @@ endif # BT
 
 config REGULATOR
 	default y if SI7210
+
+config PWM_GECKO
+	default y
+	depends on PWM
```
Comment (fkokosinski) [inline boards/arm/efr32_thunderboard/Kconfig.defconfig:63]: This seems to be redundant? `PWM_GECKO` is already being y-selected when the PWM node has the `okay` status.

File: boards/arm/efr32_thunderboard/efr32bg22_brd4184.dtsi [hunk 4]
```diff
@@ -56,3 +65,12 @@
 &sw_imu_enable {
 	enable-gpios = <&gpiob GECKO_PIN(4) GPIO_ACTIVE_HIGH>;
 };
+
+&timer0 {
+	status = "okay";
+
+	pwm0: pwm {
+		status = "okay";
+		pin-location = <GECKO_LOCATION(0) GECKO_PORT_A GECKO_PIN(4)>;
```
Comment (fkokosinski) [inline boards/arm/efr32_thunderboard/efr32bg22_brd4184.dtsi:74]: Ditto

**Review by fkokosinski — CHANGES_REQUESTED** [review]

File: boards/arm/efr32_thunderboard/Kconfig.defconfig (hunk 3 above)
Comment (wkhadgar) [reply boards/arm/efr32_thunderboard/Kconfig.defconfig:63]: Sure! I didn't notice that it was already being selected.

**Pull request closed without merging**